spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/TestSparkDistributionAndOrderingUtil.java

```diff
@@ -91,6 +105,42 @@ public void dropTable() {
     sql("DROP TABLE IF EXISTS %s", tableName);
   }
 
+  // =============================================================
+  // Distribution and ordering for write operations such as APPEND
+  // =============================================================
+  //
+  // UNPARTITIONED UNORDERED
+  // -------------------------------------------------------------------------
+  // write mode is NOT SET -> unspecified distribution + empty ordering
+  // write mode is NONE -> unspecified distribution + empty ordering
+  // write mode is HASH -> unspecified distribution + empty ordering
+  // write mode is RANGE -> unspecified distribution + empty ordering
+  //
+  // UNPARTITIONED ORDERED BY id, data
+  // -------------------------------------------------------------------------
+  // write mode is NOT SET -> ORDER BY id, data
+  // write mode is NONE -> unspecified distribution + LOCALLY ORDER BY id, data
+  // write mode is HASH -> unspecified distribution + LOCALLY ORDER BY id, data
+  // write mode is RANGE -> ORDER BY id, data
+  //
+  // PARTITIONED BY date, days(ts) UNORDERED
+  // -------------------------------------------------------------------------
+  // write mode is NOT SET -> CLUSTER BY date, days(ts) + LOCALLY ORDER BY date, days(ts)
+  // write mode is NOT SET (fanout) -> CLUSTER BY date, days(ts) + empty ordering
+  // write mode is NONE -> unspecified distribution + LOCALLY ORDERED BY date, days(ts)
+  // write mode is NONE (fanout) -> unspecified distribution + empty ordering
```

Comment:
I disable local sort by partition columns in regular writes and in CoW operations if fanout writers enabled. If the table sort order is undefined, no need to sort records by partition columns if we are not using clustered writers.

Reply:
Curious, would it be useful to have all these distribution and ordering examples accessible in our public docs?

Reply:
@amogh-jahagirdar, yes. It would be good to have these documented.

Reply:
Will add, I believe @RussellSpitzer also had some doc update PR, I am yet to check it out.